### PR TITLE
Add namespace to message element in mod_inbox

### PR DESCRIPTION
**Pull request by d1m0nh**
I wrote a plugin for stanza.io for mod_inbox. My problem was, that mod_inbox doesn`t return namespace in message element and without a namespace attribute stanza.io cannot deserialize the message correctly. I have added the namespace to the message.

### CODE REVIEW

File: src/inbox/mod_inbox_utils.erl [hunk 1]
```diff
@@ -135,3 +136,14 @@ reset_marker_to_bin(displayed) -> <<"displayed">>;
 reset_marker_to_bin(acknowledged) -> <<"acknowledged">>;
 reset_marker_to_bin(received) -> <<"received">>;
 reset_marker_to_bin(Unknown) -> throw({unknown_marker, Unknown}).
+
+-spec fill_namespace_attr(Msg :: exml:element()) -> exml:element().
+fill_namespace_attr(Msg = #xmlel{name = Name, attrs = Attrs}) when Name =:= <<"message">>  ->
```
Comment (kanes115) [inline src/inbox/mod_inbox_utils.erl]: Matching on `<<"message">>` inside `#xmlel` would shorten this line significantly and make it much more readable :)

**Comment by kanes115** [comment]
And we need a rebase with master to check if build on Travis passes

File: src/inbox/mod_inbox_utils.erl [hunk 2]
```diff
@@ -135,3 +136,14 @@ reset_marker_to_bin(displayed) -> <<"displayed">>;
...
+  case xml:get_attr(<<"xmlns">>, Attrs) of
+    false ->
```
Comment (fenek) [inline src/inbox/mod_inbox_utils.erl]: MongooseIM source code uses 4-space indentation.

File: src/inbox/mod_inbox_utils.erl [hunk 3]
```diff
@@ -135,3 +136,14 @@ reset_marker_to_bin(displayed) -> <<"displayed">>;
 reset_marker_to_bin(acknowledged) -> <<"acknowledged">>;
 reset_marker_to_bin(received) -> <<"received">>;
 reset_marker_to_bin(Unknown) -> throw({unknown_marker, Unknown}).
+
+-spec fill_namespace_attr(Msg :: exml:element()) -> exml:element().
+fill_namespace_attr(Msg = #xmlel{name = Name, attrs = Attrs}) when Name =:= <<"message">>  ->
+  case xml:get_attr(<<"xmlns">>, Attrs) of
```
Comment (fenek) [inline src/inbox/mod_inbox_utils.erl]: `xml` module is slowly phased out. Please use `exml_query` instead.

**Comment by fenek** [comment]
Hi @d1m0nh 

Is there a chance you could apply our review to your PR? :)

**Comment by fenek** [comment]
Superseded by more generic solution in #2423

**Comment by dustinswan** [comment]
Hello @d1m0nh ! Do you happen to still have your stanza.io mod_inbox plugin? I'm trying to write one myself but running into issues.

**Comment by d1m0nh** [comment]
> Hello @d1m0nh ! Do you happen to still have your stanza.io mod_inbox plugin? I'm trying to write one myself but running into issues.

Yes I still have it. You can use it
[mod_inbox.js.zip](https://github.com/esl/MongooseIM/files/8596211/mod_inbox.js.zip)
.

**Comment by dustinswan** [comment]
> Yes I still have it. You can use it

Thank you! I really appreciate it.

**Pull request closed without merging**